src/frontend/components/ProductCard/ProductCard.jsx

```diff
@@ -50,22 +65,51 @@ const ProductCard = ({
           </div>
         </div>
       </div>
-      <div className="card-btns-wrapper p-3 d-flex flex-col">
-        <button
-          type="button"
-          className="btn btn-primary rounded-sm text-sm p-3 mx-4"
-        >
-          <i className="fas fa-shopping-cart mr-3"></i>
-          Add to Cart
-        </button>
-        <button
-          type="button"
-          className="btn btn-outline btn-primary-outline rounded-sm text-sm p-3 mx-4"
-        >
-          <i className="fas fa-heart mr-3"></i>
-          Add to Wishlist
-        </button>
-      </div>
+      {page === "Products" ? (
+        <div className="card-btns-wrapper p-3 d-flex flex-col">
+          <button
+            type="button"
+            className="btn btn-primary rounded-sm text-sm p-3 mx-4"
+          >
+            <i className="fas fa-shopping-cart mr-3"></i>
+            Add to Cart
+          </button>
+          {isProductInWishlist(wishlist, _id) ? (
```

Comment:
Hey Viresh, 

Since the design of both the buttons (Add to Wishlist and Remove from Wishlist) is the same, you could perhaps use conditional rendering only to toggle the text between 'Add to Wishlist' and 'Remove from Wishlist'.
Also, you can toggle between addToWishlist function and removeFromWishlist function by using event.target.innerText 
eg. if e.target.innerText === "Add to Wishlist" ? addToWishlist() : removeFromWishlist()

Again these are just some suggestions to reduce repetition of code.
Everything else here looks good to merge